### ISSUE TITLE
Install

### ISSUE THREAD

**Issue by tofflemire** [t-brandt/orvara#55]
Hi there, running into a hiccup with the install. 

cython install went fine, but the install command returns this error (after checking all the dependencies are satisfied):

`Installing collected packages: orvara
  Running setup.py develop for orvara
    ERROR: Command errored out with exit status 1:
     command: /anaconda3/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/Users/bt9644/github/orvara/setup.py'"'"'; __file__='"'"'/Users/bt9644/github/orvara/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps
         cwd: /Users/bt9644/github/orvara/
    Complete output (81 lines):
    running develop
    running egg_info
    creating orvara.egg-info
    writing orvara.egg-info/PKG-INFO
    writing dependency_links to orvara.egg-info/dependency_links.txt
    writing entry points to orvara.egg-info/entry_points.txt
    writing requirements to orvara.egg-info/requires.txt
    writing top-level names to orvara.egg-info/top_level.txt
    writing manifest file 'orvara.egg-info/SOURCES.txt'
    reading manifest file 'orvara.egg-info/SOURCES.txt'
    writing manifest file 'orvara.egg-info/SOURCES.txt'
    running build_ext
    building 'orvara.orbit' extension
    creating build
    creating build/temp.macosx-10.9-x86_64-3.6
    creating build/temp.macosx-10.9-x86_64-3.6/orvara
    gcc -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -stdlib=libc++ -Iorvara -I/anaconda3/include/python3.6m -c orvara/orbit.c -o build/temp.macosx-10.9-x86_64-3.6/orvara/orbit.o -O3
    In file included from orvara/orbit.c:767:
    orvara/calcEA.h:172:10: error: expected identifier or '('
      double B0, B1, B2, idx;
             ^
    /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/sys/termios.h:291:17: note: expanded from macro 'B0'
    #define B0      0
                    ^
    In file included from orvara/orbit.c:767:
    orvara/calcEA.h:175:5: error: use of undeclared identifier 'idx'
        idx = 1./(bounds[i + 1] - bounds[i]);
        ^
    orvara/calcEA.h:179:53: error: use of undeclared identifier 'idx'
        B0 = idx*(-EA_tab[k + 2] - idx*(EA_tab[k + 1] - idx*pi_d_12));
                                                        ^
    orvara/calcEA.h:179:32: error: use of undeclared identifier 'idx'
        B0 = idx*(-EA_tab[k + 2] - idx*(EA_tab[k + 1] - idx*pi_d_12));
                                   ^
    orvara/calcEA.h:179:10: error: use of undeclared identifier 'idx'
        B0 = idx*(-EA_tab[k + 2] - idx*(EA_tab[k + 1] - idx*pi_d_12));
             ^
    orvara/calcEA.h:180:5: error: use of undeclared identifier 'B1'
        B1 = idx*(-2*EA_tab[k + 2] - idx*(EA_tab[k + 1] - EA_tab[k + 7]));
        ^
    orvara/calcEA.h:180:34: error: use of undeclared identifier 'idx'
        B1 = idx*(-2*EA_tab[k + 2] - idx*(EA_tab[k + 1] - EA_tab[k + 7]));
                                     ^
    orvara/calcEA.h:180:10: error: use of undeclared identifier 'idx'
        B1 = idx*(-2*EA_tab[k + 2] - idx*(EA_tab[k + 1] - EA_tab[k + 7]));
             ^
    orvara/calcEA.h:181:5: error: use of undeclared identifier 'B2'
        B2 = idx*(EA_tab[k + 8] - EA_tab[k + 2]);
        ^
    orvara/calcEA.h:181:10: error: use of undeclared identifier 'idx'
        B2 = idx*(EA_tab[k + 8] - EA_tab[k + 2]);
             ^
    orvara/calcEA.h:183:28: error: use of undeclared identifier 'B1'
        EA_tab[k + 3] = B2 - 4*B1 + 10*B0;
                               ^
    orvara/calcEA.h:183:21: error: use of undeclared identifier 'B2'
        EA_tab[k + 3] = B2 - 4*B1 + 10*B0;
                        ^
    orvara/calcEA.h:184:25: error: use of undeclared identifier 'B2'
        EA_tab[k + 4] = (-2*B2 + 7*B1 - 15*B0)*idx;
                            ^
    orvara/calcEA.h:184:32: error: use of undeclared identifier 'B1'
        EA_tab[k + 4] = (-2*B2 + 7*B1 - 15*B0)*idx;
                                   ^
    orvara/calcEA.h:184:44: error: use of undeclared identifier 'idx'
        EA_tab[k + 4] = (-2*B2 + 7*B1 - 15*B0)*idx;
                                               ^
    orvara/calcEA.h:185:29: error: use of undeclared identifier 'B1'
        EA_tab[k + 5] = (B2 - 3*B1 + 6*B0)*idx*idx;
                                ^
    orvara/calcEA.h:185:22: error: use of undeclared identifier 'B2'
        EA_tab[k + 5] = (B2 - 3*B1 + 6*B0)*idx*idx;
                         ^
    orvara/calcEA.h:185:40: error: use of undeclared identifier 'idx'
        EA_tab[k + 5] = (B2 - 3*B1 + 6*B0)*idx*idx;
                                           ^
    orvara/calcEA.h:185:44: error: use of undeclared identifier 'idx'
        EA_tab[k + 5] = (B2 - 3*B1 + 6*B0)*idx*idx;
                                               ^
    19 errors generated.
    error: command 'gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /anaconda3/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/Users/bt9644/github/orvara/setup.py'"'"'; __file__='"'"'/Users/bt9644/github/orvara/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps Check the logs for full command output.
`

Apologies for the long read out. It looks like it is a problem with gcc. Here is the version I am working with:
bash-3.2$ gcc --version
Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/c++/4.2.1
Apple LLVM version 10.0.1 (clang-1001.0.46.4)
Target: x86_64-apple-darwin18.7.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin

Let me know if there is any other info I can provide to help diagnose this. 

Thanks in advance!

**Comment by gmbrandt** [comment]
Hmm, @Yiting206265 or @t-brandt , any input on installing on mac? 
I know this came up during the sagan workshop with gcc versions etc.

**Comment by tofflemire** [comment]
Update on my end:

I created a conda virtual environment (called orvara) and retried the install. This resulted in the same error as above. Digging through some of the other issues, I tired to update the gcc with the following:
`(orvara) bash-3.2$ conda install --name orvara -c conda-forge gcc`

Now I am getting this error: 
Installing collected packages: orvara                                                                                  
  Running setup.py develop for orvara                                                                                  
    ERROR: Command errored out with exit status 1:                                                                     
     command: /anaconda3/envs/orvara/bin/python -c 'import sys, setuptools, toke                                       nize; sys.argv[0] = '"'"'/Users/bt9644/github/orvara/setup.py'"'"'; __file__='"'                                       "'/Users/bt9644/github/orvara/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"',                                        open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();ex                                       ec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps                                                         
         cwd: /Users/bt9644/github/orvara/                                                                             
    Complete output (26 lines):                                                                                        
    running develop                                                                                                    
    running egg_info
    creating orvara.egg-info
    writing orvara.egg-info/PKG-INFO
    writing dependency_links to orvara.egg-info/dependency_links.txt
    writing entry points to orvara.egg-info/entry_points.txt
    writing requirements to orvara.egg-info/requires.txt
    writing top-level names to orvara.egg-info/top_level.txt
    writing manifest file 'orvara.egg-info/SOURCES.txt'
    reading manifest file 'orvara.egg-info/SOURCES.txt'
    writing manifest file 'orvara.egg-info/SOURCES.txt'
    running build_ext
    building 'orvara.orbit' extension
    creating build
    creating build/temp.macosx-10.9-x86_64-3.6
    creating build/temp.macosx-10.9-x86_64-3.6/orvara
    gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/anaconda3/envs/orvara/include -arch x86_64 -I/anaconda3/envs/orvara/include -arch x86_64 -Iorvara -I/anaconda3/envs/orvara/include/python3.6m -c orvara/orbit.c -o build/temp.macosx-10.9-x86_64-3.6/orvara/orbit.o -O3
    In file included from /anaconda3/envs/orvara/lib/gcc/x86_64-apple-darwin11.4.2/4.8.5/include-fixed/syslimits.h:7:0,
                     from /anaconda3/envs/orvara/lib/gcc/x86_64-apple-darwin11.4.2/4.8.5/include-fixed/limits.h:34,
                     from /anaconda3/envs/orvara/include/python3.6m/Python.h:11,
                     from orvara/orbit.c:26:
    /anaconda3/envs/orvara/lib/gcc/x86_64-apple-darwin11.4.2/4.8.5/include-fixed/limits.h:168:61: fatal error: limits.h: No such file or directory
     #include_next <limits.h>  /* recurse down to the real one */
                                                                 ^
    compilation terminated.
    error: command 'gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /anaconda3/envs/orvara/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/Users/bt9644/github/orvara/setup.py'"'"'; __file__='"'"'/Users/bt9644/github/orvara/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps Check the logs for full command output.

This seems like progress, but I'm still not sure how best to proceed. 

Thanks!

**Comment by gmbrandt** [comment]
I do not have a mac, so this is my best guess of what may help:
When we install this, we run `pip install -e .`  the -e flag enables developer mode, and allows the c files to  be built. It is required.

I am not familiar with `conda install --name orvara -c conda-forge gcc` 
is there an analogous way to use the -e flag here?

Also followup, are you first downloading the source files to your local computer, cd'ing into that directory, then installing from the root directory? If so, you want to make sure you are not pointing the installer to the subfolder "orvara", but rather to the directory you are currently in (i.e., the root directory for the package. This directory _contains_ the subfolder orvara).

**Comment by tofflemire** [comment]
Okay, I think I have it sorted.

Two things changed, which made the install and tests successful. 

1) I ran a mac update, still Mojave 10.14.6, so far from the most up to date, but now current within this version.

2) My virtual environment was defaulting to python 3.6. I made a new environment with python3.8.

I would guess that the latter made the difference, but can't be sure. If the python3.8 version tracks with your experience, it may be worth adding that to the install documentation.
